homeassistant/helpers/entity.py

```diff
@@ -238,7 +251,14 @@ def async_update_ha_state(self, force_refresh=False):
 
     def remove(self) -> None:
         """Remove entitiy from HASS."""
-        self.hass.states.remove(self.entity_id)
+        run_callback_threadsafe(self.hass.loop, self.async_remove)
+
+    def async_remove(self) -> None:
```

Comment:
We should make this a coroutine. We don't know what in the future people might implement for this.